### DISCUSSION TITLE
Env variables in vercel.json

### DISCUSSION THREAD

**Discussion by JotaOdiceu** [vercel/community#5607] [Help]
### Summary

Hello guys, I'm setting up an API in Vercel and I need to set the HTTP headers according to the environment, whether it's dev or prod, I've tried some alternatives but I haven't been successful, has anyone gone through something similar?

### Example

Basic Fastify API

### Steps to Reproduce

```
{
  "version": 2,
  "builds": [
    {
      "src": "./build/server.js",
      "use": "@now/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "./build/server.js",
      "methods": ["GET", "POST", "PUT", "DELETE", "PATCH", "OPTIONS"],
      "headers": {
        "Access-Control-Allow-Origin": "@CORS_ORIGIN",
        "Access-Control-Allow-Credentials": "true"
      }
    }
  ]
}
```

I already tried using $CORS_ORIGIN, but without success.

**Comment by Felt-Gaming-Labs** [comment]
Try the following:

**Setting Environment Variables in Vercel:**

Go to your Vercel dashboard.
Navigate to your project settings.
Add environment variables: DEV_CORS_ORIGIN and PROD_CORS_ORIGIN with the respective URLs.

**Modifying Fastify Server:**

Fastify provides a way to handle CORS. We will use the environment variables to dynamically set the origin option in Fastify's CORS plugin.
Here's an example of how to adjust your Fastify server:

const fastify = require('fastify')();

// Determine the environment and set the CORS origin
const CORS_ORIGIN = process.env.NODE_ENV === 'production' ? process.env.PROD_CORS_ORIGIN : process.env.DEV_CORS_ORIGIN;

fastify.register(require('fastify-cors'), { 
  origin: CORS_ORIGIN,
  credentials: true
});

// Your Fastify routes and logic here

const start = async () => {
  try {
    await fastify.listen(process.env.PORT || 3000, '0.0.0.0');
    console.log(`Server listening`);
  } catch (err) {
    fastify.log.error(err);
    process.exit(1);
  }
};
start();

**Adjusting Vercel Configuration (vercel.json):**

Ensure that your vercel.json is correctly set up for Fastify. The @now/node builder is deprecated and should be replaced with @vercel/node.
Here’s a revised vercel.json:

json

{
  "version": 2,
  "builds": [
    {
      "src": "./build/server.js",
      "use": "@vercel/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "/build/server.js",
      "methods": ["GET", "POST", "PUT", "DELETE", "PATCH", "OPTIONS"]
    }
  ]
}


Ensure that the build/server.js path correctly points to your built Fastify server file.
Test both your development and production environments to verify that the CORS headers are being set correctly.
If you're deploying via Git, remember to add and commit these changes before deploying to Vercel.
Fastify API on Vercel should correctly set HTTP headers based on the deployment environment.

**Reply by LeoAnt02** [reply]
How can we achieve this using a simple react app with vercel ? 

We tried the following 

- using @ instead of ${}
- editing the vercel.json during build 

none of it work
